src/engine/Account-Imap.cpp

```diff
@@ -0,0 +1,1186 @@
+GncImportMatchMap::get_account()
+{
+std::cout << "get_account" << std::endl;
+    return account;
+}
+
+QofBook *
+GncImportMatchMap::get_book()
+{
+std::cout << "get_book" << std::endl;
+    return book;
+}
+
+Account *
+GncImportMatchMap::find_account (const char* category, const char *key)
+{
+    KvpValue *value;
+
+std::cout << "find_account " << category << ", " << key << std::endl;
+
+    if (!key) return nullptr;
+
+    auto slots = qof_instance_get_slots(QOF_INSTANCE(account));
+
+    if (!category)
+        value = slots->get_slot({IMAP_FRAME, key});
+    else
+        value = slots->get_slot({IMAP_FRAME, category, key});
+
+    if (value == nullptr || value->get_type() != KvpValue::Type::GUID)
+       return nullptr;
+
+    return xaccAccountLookup(value->get<GncGUID*>(), book);
+}
+
+void
+GncImportMatchMap::add_account (const char *category, const char *key, Account *acc)
+{
+    if (!key || !acc || (strlen (key) == 0)) return;
+
+std::cout << "add_account " << category << ", " << key << std::endl;
+
+    auto acc_slots = qof_instance_get_slots(QOF_INSTANCE(account));
+
+    auto acc_val = new KvpValue(const_cast<GncGUID*>(xaccAccountGetGUID(acc)));
+
+    xaccAccountBeginEdit (account);
+
+    if (!category)
+        acc_slots->set_path({IMAP_FRAME, key}, acc_val);
+    else
+        acc_slots->set_path({IMAP_FRAME, category, key}, acc_val);
+
+    qof_instance_set_dirty (QOF_INSTANCE (account));
+    xaccAccountCommitEdit (account);
+}
+
+void
+GncImportMatchMap::delete_account (const char *category, const char *key)
+{
+    std::string kvp_path;
+    std::string delim = "/";
+
+    if ( !key) return;
+
+std::cout << "Delete_account" << std::endl;
+
+    std::string str_key (key);
+
+    if (!category)
+        kvp_path = IMAP_FRAME + delim + str_key;
+    else
+    {
+        std::string str_category (category);
+        kvp_path = IMAP_FRAME + delim + str_category + delim + str_key;
+    }
+    xaccAccountBeginEdit (account);
+
+    if (qof_instance_has_slot (QOF_INSTANCE (account), kvp_path.c_str()))
+    {
+        qof_instance_slot_delete (QOF_INSTANCE (account), kvp_path.c_str());
+
+        if (category)
+        {
+            std::string str_category (category);
+            kvp_path = IMAP_FRAME + delim + str_category;
+
+            qof_instance_slot_delete_if_empty (QOF_INSTANCE (account), kvp_path.c_str());
+        }
+        qof_instance_slot_delete_if_empty (QOF_INSTANCE (account), IMAP_FRAME);
+    }
+    qof_instance_set_dirty (QOF_INSTANCE (account));
+    xaccAccountCommitEdit (account);
+}
+
+Account *
+GncImportMatchMap::find_account_bayes (GList* tokens)
+{
+std::cout << "find_account_bayes" << std::endl;
+    return nullptr;
+}
+
+void
+GncImportMatchMap::add_account_bayes (GList* tokens, Account *acc)
+{
+std::cout << "add_account_bayes" << std::endl;
+}
+
+/*****************************************************************************/
+
+// C access functions
+#ifdef __cplusplus
+extern "C" {
+#endif
+
```

Comment:
You only do that in the header. That takes care of setting C linkage for the declared functions, no need to do it in the implementation file.